wagtail/models/__init__.py

```diff
@@ -2136,15 +2136,16 @@ def get_url_parts(self, request=None):
         if not possible_sites:
             return None
 
+        unique_site_ids = {values[0] for values in possible_sites}
         site_id, root_path, root_url, language_code = possible_sites[0]
 
-        site = Site.find_for_request(request)
-        if site:
-            for site_id, root_path, root_url, language_code in possible_sites:
-                if site_id == site.pk:
-                    break
-            else:
-                site_id, root_path, root_url, language_code = possible_sites[0]
```

Comment:
FYI: I know I've changed more than I needed to here, but I always found these lines very difficult to follow